src/fides/ctl/core/api.py

```diff
@@ -1,9 +1,35 @@
 """A wrapper to make calling the API consistent across fides."""
-from typing import Dict, List
+from types import ModuleType
+from typing import Any, Callable, Dict, List, Union
 
 import requests
+from starlette.testclient import TestClient
 
 from fides.api.ctl.routes.util import API_PREFIX
+from fides.ctl.core.config import get_config
+
+CONFIG = get_config()
+
+
+def verify_client(func: Callable) -> Callable:
```

Comment:
Directly mixing in test logic with code logic feels suboptimal to me, I think there might be another solution.

I agree that passing in the client feels like a good move here, maybe there is some kind of mocking/testing library specifically for `requests`, so the application code never has to know if it is using a test client or a normal client?

I found the following SO thread that has some potentially useful solutions:

https://stackoverflow.com/questions/9559963/unit-testing-a-python-app-that-uses-the-requests-library

Reply:
I might also be totally off-base here and not grasping that potentially the `TestClient` can't be replaced by another library

Reply:
> Directly mixing in test logic with code logic feels suboptimal to me

Agreed, this was a last resort given what I understood to be the constraints! In particular, it sounds like the CLI needs to make API requests


> I found the following SO thread that has some potentially useful solutions:
https://stackoverflow.com/questions/9559963/unit-testing-a-python-app-that-uses-the-requests-library

hmm, I'm hesitant to mock the response from the `requests` calls here, because then we'd be mocking most of the logic used by the CLI


> I might also be totally off-base here and not grasping that potentially the TestClient can't be replaced by another library

The core of the issue is that I want to unit test the calls in `src/fides/ctl/core/api.py` which are primarily used by the CLI (`test_cli`, `test_api`, `test_api_helpers`).  You said above the CLI should always be making [API requests against the webserver ](https://github.com/ethyca/fides/pull/1731#discussion_r1019989425) because there's no guarantee that it is running in the same place as the server.

The `TestClient` doesn't actually make http requests, it simulates making http requests and calls functions directly.  This is great for testing our routes: you don't need a live webserver, you reduce overhead of going through an entire http stack, etc.  

However, on this page, I'm trying to unittest locations which make http requests to our webserver.   I can't inject the test client here, it's not an endpoint per se, and I can't create a client in this context without getting multiple event loops running.  The only workable option I've seen so far is to pass in the TestClient for use in pytest, otherwise, it makes http calls. Open to other ideas!.

Reply:
Discussed in person - Thomas suggested monkeypatching https://docs.pytest.org/en/7.1.x/how-to/monkeypatch.html so I'll investigate that next!